Add support for new API module org.graalvm.nativeimage.libgraal

https://github.com/oracle/graal/pull/10380 introduced a new API module - part of native-image, which a `libgraal` build would be able to use. Mandrel doesn't build `libgraal`. That remains true after this refactoring.

Currently a Mandrel build of revision 8e0b9db6a2b96c5c6584480bafa42515f3d607cd with `JDK-25+9` results in this build failure when trying to build the native image agents:

```
Error occurred during initialization of boot layer
java.lang.module.FindException: Module org.graalvm.nativeimage.libgraal not found, required by org.graalvm.nativeimage.base
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: Failed, exit code: 1
	at OperatingSystem.exec(build.java:1876)
	at build.buildAgents(build.java:306)
	at build.main(build.java:249)
Caused by: java.lang.RuntimeException: Failed, exit code: 1
	at OperatingSystem.exec(build.java:1865)
	at build.buildAgents(build.java:306)
	at build.main(build.java:249)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:565)
	at jdk.compiler/com.sun.tools.javac.launcher.SourceLauncher.execute(SourceLauncher.java:254)
	at jdk.compiler/com.sun.tools.javac.launcher.SourceLauncher.run(SourceLauncher.java:138)
	at jdk.compiler/com.sun.tools.javac.launcher.SourceLauncher.main(SourceLauncher.java:76)
Error: Process completed with exit code 1.
```

The new `org.graalvm.nativeimage.libgraal` jar file - that has minimal dependencies - gets built as part of the Mandrel build already, but isn't copied to the Mandrel installation:

```
$ jar -d -f ./sdk/mxbuild/dists/nativeimage-libgraal.jar
org.graalvm.nativeimage.libgraal jar:file:///disk/graal/upstream-sources/graal/./sdk/mxbuild/dists/nativeimage-libgraal.jar!/module-info.class
exports org.graalvm.nativeimage.libgraal
exports org.graalvm.nativeimage.libgraal.hosted
requires java.base mandated
requires org.graalvm.nativeimage transitive
requires org.graalvm.word transitive
qualified exports org.graalvm.nativeimage.libgraal.impl to org.graalvm.nativeimage.builder
```

--- a/build.java
+++ b/build.java
@@ -976,6 +976,8 @@ class Mx
                 new Path[]{sdkDistPath.resolve("collections.jar"), Path.of("lib", "jvmci", "collections.jar")}),
             new SimpleEntry<>("org.graalvm.sdk:word.jar",
                 new Path[]{sdkDistPath.resolve("word.jar"), Path.of("lib", "jvmci", "word.jar")}),
+            new SimpleEntry<>("org.graalvm.nativeimage:native-libgraal.jar",
+                    new Path[]{sdkDistPath.resolve("nativeimage-libgraal.jar"), Path.of("lib", "jvmci", "nativeimage-libgraal.jar")}),
             new SimpleEntry<>("org.graalvm.nativeimage:svm.jar",
                 new Path[]{substrateDistPath.resolve("svm.jar"), Path.of("lib", "svm", "builder", "svm.jar")}),
             new SimpleEntry<>("org.graalvm.nativeimage:native-image-base.jar",

